sstables/sstables.cc

```diff
@@ -2594,6 +2599,7 @@ future<checksum> sstable::read_checksum() {
 
         co_await crc_stream.close();
         maybe_rethrow_exception(std::move(ex));
+        _components->checksum = checksum->weak_from_this();
     });
 
     co_return checksum;
```

Comment:
minor nit: `co_return std::move(checksum);`

Unlike `return`, which has copy-ellision, `co_return x` is syntactic sugar for `future<>::set_value(x)`, so `co_return std::move(x)` translates to `future<>::set_value(std::move(x))` avoiding a copy that would happen with a simple `co_return x;`.